### ISSUE TITLE
Misalignment of data augmentation BGR in guide doc and real code

### ISSUE THREAD

**Issue by shinkyo0513** [ultralytics/ultralytics#23695]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Augmentation

### Bug

Based on the code of augment.py in line 2133 
`img = np.ascontiguousarray(img[::-1] if random.uniform(0, 1) > self.bgr and img.shape[0] == 3 else img)`, the img channel is flipped when the random number larger than self.bgr. 
This means, **if self.bgr is 0.0, then the img channel will be almost always flipped. And when self.bgr is 1.0, the flip will never happen.**

But in the guide doc of "[Data Augmentation using Ultralytics YOLO](https://docs.ultralytics.com/guides/yolo-data-augmentation/#flip-left-right-fliplr:~:text=fliplr%20on-,BGR%20Channel%20Swap%20(bgr),has%20a%2050%25%20chance%20of%20being%20converted%20from%20RGB%20to%20BGR.,-Purpose%3A%20Increases%20robustness)", `with bgr=1.0 ensuring all images undergo the channel swap and bgr=0.0 disabling it`. This is inverse to the logic in code.

### Environment

Ultralytics 8.4.14

### Minimal Reproducible Example

We do not need to run the code for knowing the misalignment.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @shinkyo0513, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for flagging this potential misalignment between the docs and `augment.py` for the BGR channel swap behavior 🧐📄. This is an automated response to help us triage quickly, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Even when an issue appears “logic-only”, an MRE is still important so we can validate the exact pipeline/codepath, config values, and image tensor layout being used in your environment.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The image is loaded as BGR by OpenCV. The channel swap referred to in the docs is from RGB to BGR. Not BGR to RGB. When `self.bgr` is 0, it always swaps in the code to convert from BGR to RGB. When it's 1, it stays in BGR. It needs to be inverse because the loaded image is in BGR, not RGB.

**Comment by shinkyo0513** [comment]
Thanks for the explanation, that clarifies the intended behavior of the code, and I agree the implementation itself is internally consistent given OpenCV loads images in BGR.

My concern is mainly about documentation semantics and user expectation, not the code logic.

**In the docs, bgr is described as “channel swap”** with examples like:

- bgr=1.0 ensures all images undergo the channel swap
- bgr=0.0 disables it

**This wording implicitly assumes the input image is in RGB, while in reality the code assumes the input is BGR** and bgr controls the probability of keeping BGR vs converting to RGB. So from a user perspective, the meaning of the parameter is effectively inverted.

This becomes especially problematic in cases where:

- images are not loaded via OpenCV (e.g. PIL, torchvision, custom dataloaders that produce RGB),
- the input is non-standard or multimodal (e.g. grayscale stacked with extra channels, event frames, sensor data),

users override the dataloader and still rely on Ultralytics augmentations.

In those cases, enabling bgr as described in the docs can silently corrupt channel semantics, because the augmentation implicitly assumes BGR input.

I think this would be much clearer and safer if the docs explicitly stated something like:

> Images are loaded in BGR order by OpenCV.
> bgr controls the probability of keeping BGR instead of converting to RGB.

Or alternatively:

- rename the parameter to something like keep_bgr_prob, or
- redefine bgr as “probability to swap channels” and invert the condition in code.

My main goal is to avoid subtle bugs for users who don’t strictly follow the OpenCV BGR pipeline.

**Comment by Y-T-G** [comment]
The docs is correct because it's describing the end behavior of the augmentation, not what the code specifically does. What the code does is part of the docstring. The augmentation docs is intended to describe the effect of augmentation parameters and not the code.

`bgr` is indeed a channel swap because the original image is RGB. So `bgr=1` performs channel swap and `bgr=0`. The difference in the code comes from OpenCV loading the image as BGR instead of RGB.  It's more confusing to say `bgr=0` to channel swap when the image stays in RGB with `bgr=0`. The docs is not intended to confuse the users with technical details.
